pandas/tests/io/formats/test_to_csv.py

```diff
@@ -561,3 +563,21 @@ def test_to_csv_compression(self, compression_only,
             result = pd.read_csv(path, index_col=0,
                                  compression=read_compression)
             tm.assert_frame_equal(result, df)
+
+    @pytest.mark.parametrize("df_value", [float('nan')])
+    @pytest.mark.parametrize("df_new_type", ["Int64"])
+    def test_to_csv_na_rep_long_string(self, df_value, df_new_type):
+        # see gh-25099
+        df = pd.DataFrame({"c": [df_value] * 3})
+        df = df.astype(df_new_type)
+        expected_rows = ['c',
+                         'mynull',
+                         'mynull',
+                         'mynull']
+        expected = tm.convert_rows_list_to_csv_str(expected_rows)
+
+        csv_data = StringIO()
+        df.to_csv(csv_data, index=False, na_rep='mynull', encoding='ascii')
```

Comment:
Actually you don't even need io - just don't pass an argument for filepath_or_buffer and you should get a string back

Reply:
Brilliant!!!